### PR TITLE
Updated the pytorch examples to use pytorch 1.9.0 and pytorch-lightning 1.4.0

### PR DESCRIPTION
**Pull request by KasirajanA**
Signed-off-by: Kasirajan Alayappan <kasirajan.alayappan@ideas2it.com>

## What changes are proposed in this pull request?

Updated the pytorch examples to use pytorch 1.9.0 and latest pytorch-lightning

## How is this patch tested?

Ran the examples locally

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow TorchServe Deployment Plugin users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?
Components
- [ ] `area/deploy`: Main deployment plugin logic
- [ ] `area/build`: Build and test infrastructure for MLflow TorchServe Deployment Plugin
- [ ] `area/docs`: MLflow TorchServe Deployment Plugin documentation pages
- [x] `area/examples`: Example code

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: examples/MNIST/conda.yaml [hunk 1]
```diff
@@ -11,3 +11,4 @@ dependencies:
   - matplotlib
   - pytorch-lightning>=1.2.3
   - mlflow>=1.14.0
+  - captum
```
Comment (chauhang) [inline examples/MNIST/conda.yaml:14]: Please update the PTL version as well

File: examples/MNIST/mnist_model.py [hunk 2]
```diff
@@ -13,9 +13,13 @@
 import mlflow.pytorch
 import pytorch_lightning as pl
 import torch
-from pytorch_lightning.overrides.data_parallel import (
-    LightningDistributedDataParallel,
-    LightningDataParallel,
+# from pytorch_lightning.overrides.data_parallel import (
```
Comment (chauhang) [inline examples/MNIST/mnist_model.py:16]: What is the reason for this change?

**Review by chauhang — CHANGES_REQUESTED** [review]
Please see comments inline

File: examples/MNIST/mnist_model.py (hunk 2 above)
Comment (KasirajanA) [reply examples/MNIST/mnist_model.py:16]: @chauhang, both lightningdataparallel and dataparallel is removed in latest pytorch lightning and they suggested to directly use torch.nn.parallel.DistributedDataParallel and torch.nn.parallel.DataParallel.

Reference: https://github.com/PyTorchLightning/pytorch-lightning/pull/7510/files

File: examples/IrisClassificationTorchScript/iris_datamodule.py [hunk 3]
```diff
@@ -11,21 +12,25 @@ def __init__(self):
...
+            iris = load_iris()
+            df = iris.data
+            target = iris["target"]
+
+            data = torch.Tensor(df).float()
+            labels = torch.Tensor(target).long()
+            RANDOM_SEED = 42
+            seed_everything(RANDOM_SEED)
+
+            data_set = TensorDataset(data, labels)
```
Comment (chauhang) [inline examples/IrisClassificationTorchScript/iris_datamodule.py:32]: What is the reason for moving this from "prepare_data" to "setup" method?

Comment (shrinath-suresh) [reply examples/IrisClassificationTorchScript/iris_datamodule.py:32]: There is a change in behaviour from PTL `1.3.x` and ` 1.4.x`. 

`prepare_data` method does not return anything. Since the function calls are controlled by PTL DataModule class, prepare_data is getting called and the dataset is not passed on to the setup (as prepare_data doesnt return anything).

However, in the older version it is not the case. The behaviour can be observed in screenshots

PTL 1.3.x:

```
Returning from prepare data:  <torch.utils.data.dataset.TensorDataset object at 0x7f41043eb6a0>
Received in setup :  <torch.utils.data.dataset.TensorDataset object at 0x7f41043eb6a0>

```
PTL 1.4.x:
```
Returning from prepare data:  <torch.utils.data.dataset.TensorDataset object at 0x7fa58f2d1af0>
Received in setup :  None

```
 
image
image


Since the data preparation is simple in iris classification, @KasirajanA  has moved the logic to setup (instead of introducing another class variable).

**Review by chauhang — APPROVED** [review]
